sql/core/src/main/scala/org/apache/spark/sql/execution/python/PythonUDF.scala

```diff
@@ -36,9 +36,12 @@ case class PythonUDF(
     broadcastVars: java.util.List[Broadcast[PythonBroadcast]],
     accumulator: Accumulator[java.util.List[Array[Byte]]],
     dataType: DataType,
-    children: Seq[Expression]) extends Expression with Unevaluable with Logging {
+    children: Seq[Expression])
+  extends Expression with Unevaluable with NonSQLExpression with Logging {
 
-  override def toString: String = s"PythonUDF#$name(${children.mkString(",")})"
+  override def toString: String = s"PythonUDF#$name(${children.mkString(", ")})"
 
   override def nullable: Boolean = true
+
+  override def sql: String = s"$name(${children.mkString(", ")})"
```

Comment:
Well, it does. Because we can't generate SQL string for a query plan containing a Python UDF. `SQLBuilder` tries to find `NonSQLExpression`s for this detection.